text/0000-rust-lang-crates.md

```diff
@@ -0,0 +1,231 @@
+promoted into rust-lang proper. To do so, an RFC must be written outlining the
+motivation for the crate, the reasons that community ownership are important,
+and delving into the API design and its rationale design. These RFCs are
+intended to follow similar lines to the pre-1.0 stabilization RFCs for the
+standard library (such as
+[collections](https://github.com/rust-lang/rfcs/pull/235) or
+[Duration](https://github.com/rust-lang/rfcs/pull/1040)) -- which have been very
+successful in improving API design prior to stabilization. Once a "1.0 RFC" is
+approved by the libs team, the crate moves into the rust-lang organization, and
+is henceforth governed by the whole Rust community. That means in particular
+that significant changes (certainly those that would require a major version
+bump, but other substantial PRs as well) are reviewed by the library subteam and
+may require an RFC. On the other hand, the community has broadly agreed to
+maintain the library in perpetuity (unless it is later deprecated). And again,
+as we'll see below, the promoted crate is very visibly advertised as part of the
+"core Rust" package.
+
+Promotion to 1.0 requires first-class support on all first-tier platforms,
+except for platform-specific libraries.
+
+Crates in rust-lang may issue new major versions, just like any other crates,
+though such changes should go through the RFC process. While the library subteam
+is responsible for major decisions about the library after 1.0, its original
+author(s) will of course wield a great deal of influence, and their objections
+will be given due weight in the consensus process.
+
+### Relation to `std`
+
+In many cases, the above description of the crate lifecycle is complete. But
+some rust-lang crates are destined for std. Usually this will be clear up front.
+
+When a std-destined crate has reached sufficient maturity, the libs subteam can
+call a "final comment period" for moving it into `std` proper. Assuming there
+are no blocking objections, the code is moved into `std`, and the original repo
+is left intact, with the following changes:
+
+- a minor version bump,
+- *conditionally* replacing all definitions with `pub use` from `std` (which
+  will require the ability to `cfg` switch on feature/API availability -- a
+  highly-desired feature on its own).
+
+By re-routing the library to `std` when available we provide seamless
+compatibility between users of the library externally and in `std`. In
+particular, traits and types defined in the crate are compatible across either
+way of importing them.
+
+### Deprecation
+
+At some point a library may become stale -- either because it failed to make it
+out of the nursery, or else because it was supplanted by a superior library. The
+libs subteam can deprecate nursery crates at any time, and can deprecate
+rust-lang crates through an RFC. This is expected to be a rare occurrence.
+
+Deprecated crates move to rust-lang-deprecated and are subsequently minimally
```

Comment:
It seems like we should always prefer to transfer to someone who's willing to maintain it, and only use rust-lang-deprecated as a last resort.